src/mono/mono/mini/llvmonly-runtime.c

```diff
@@ -215,6 +215,29 @@ mini_llvmonly_add_method_wrappers (MonoMethod *m, gpointer compiled_method, gboo
 	return addr;
 }
 
+static inline MonoVTableEEData*
+get_vtable_ee_data (MonoVTable *vtable)
+{
+	MonoVTableEEData *ee_data = (MonoVTableEEData*)vtable->ee_data;
+
+	if (G_UNLIKELY (!ee_data)) {
+		ee_data = m_class_alloc0 (vtable->klass, sizeof (MonoVTableEEData));
+		mono_memory_barrier ();
+		vtable->ee_data = ee_data;
+	}
+	return ee_data;
+}
+
+static void
+alloc_gsharedvt_vtable (MonoVTable *vtable)
```

Comment:
I think there is some concurrency fishiness here. Callers of this method probably expect `vtable->ee_data->gsharedvt_vtable` to have been created, however it seems possible that in the meantime the interpreter could just replace the `vtable->ee_data` resulting in a NULL gsharedvt table.